sc-tools/sc-server/src/sc-server-impl/sc-memory-json/sc-memory-json-event/sc_memory_json_events_handler.cpp

```diff
@@ -7,41 +7,62 @@
 #include "sc_memory_json_events_handler.hpp"
 
 ScMemoryJsonEventsHandler::ScMemoryJsonEventsHandler(ScServer * server)
+  : ScMemoryJsonHandler(server)
 {
   m_context = new ScMemoryContext("sc-json-socket-events-handler");
   m_manager = ScMemoryJsonEventsManager::GetInstance();
+}
 
-  m_server = server;
+ScMemoryJsonEventsHandler::~ScMemoryJsonEventsHandler()
+{
+  m_context->Destroy();
+  delete m_context;
 }
 
-std::string ScMemoryJsonEventsHandler::Handle(
+ScMemoryJsonPayload ScMemoryJsonEventsHandler::HandleRequestPayload(
     ScServerConnectionHandle const & hdl,
-    std::string const & requestMessageText)
+    std::string const & requestType,
+    ScMemoryJsonPayload const & requestPayload,
+    ScMemoryJsonPayload & errorsPayload,
+    sc_bool & status,
+    sc_bool & isEvent)
 {
-  ScMemoryJsonPayload const & requestMessage = ScMemoryJsonPayload::parse(requestMessageText);
-  ScMemoryJsonPayload requestPayload = requestMessage["payload"];
+  status = SC_FALSE;
+  isEvent = SC_TRUE;
 
   ScMemoryJsonPayload responsePayload;
-  if (requestPayload["create"].is_null() == SC_FALSE)
+  if (requestPayload.find("create") != requestPayload.cend())
+  {
     responsePayload = HandleCreate(hdl, requestPayload["create"]);
-  else if (requestPayload["delete"].is_null() == SC_FALSE)
+    status = SC_TRUE;
+  }
+  else if (requestPayload.find("delete") != requestPayload.cend())
+  {
     responsePayload = HandleDelete(hdl, requestPayload["delete"]);
+    status = SC_TRUE;
```

Comment:
Any checks for `errors` field or exception catches? I am not sure if this is a relevant field for an event handler, but it sounds like `HandleCreate()` could write errors or throw exceptions on the way, and status would get overwritten by this. Same thing across other methods in this file